world/traits.py

```diff
@@ -0,0 +1,701 @@
+# -*- coding: utf-8 -*-
+"""
+Traits Module
+
+`Trait` classes represent modifiable traits on objects or characters. They
+are instantiated by a `TraitFactory` object, which is typically set up
+as a property on the object or character's typeclass.
+
+Data persistence for `Trait` objects is handled by passing a dict from
+a typeclass instance's `db` attribute handler loaded with the appropriate
+keys to the factory which map to properties of the eventual `Trait` object.
+See _Trait Configuration_ below for more details.
+
+**Setup**
+    To use traits on an object, add a dict containing trait configuration
+    keys to the object's `db` property in the `at_object_creation` hook,
+    and define a property on the object's class that passes that Attribute
+    into the constructor and returns a `TraitFactory`.
+
+Example:
+    ```python
+    from world.traits import TraitFactory
+        ...
+    class Object(DefaultObject):
+        ...
+        def at_object_creation(self):
+            self.db.traits = {
+                # static trait example
+                'str': {'name': 'Strength',
+                        'type': 'static',
+                        'base': 5},
+                # counter trait example
+                'carry': {'name': 'Carry Weight',
+                          'type': 'counter',
+                          'base': 0,
+                          'min': 0,
+                          'max': None},
+                # gauge trait example
+                'hp': {'name': 'HP',
+                       'type': 'gauge',
+                       'base': 10}
+            }
+        ...
+        @property
+        def traits(self):
+            return TraitFactory(self.db.traits)
+    ```
+
+**Trait Configuration**
+    `Trait` objects can be configured as one of three basic types with
+    increasingly complex behavior.
+
+    * Static - A simple trait model with a base value and optional modifier.
+
+    * Counter - Trait with a base value and a modifiable current value that
+        can vary along a range defined by optional min and max values.
+
+    * Gauge - Modified counter type modeling a refillable "gauge".
+
+    All traits have a read-only `actual` property that will report the trait's
+    actual value. In addition to normal attribute notation, `Trait` objects
+    support the unary `+` operator as a shortcut to access this property.
+
```

Comment:
As we discussed in IRC the other day, the implementation of the `+` operator and other comparison/arithmetic magic methods are not really crucial to the functioning of the Trait class, but I put them there because I personally prefer the briefer syntax that they allow over the more explicit and verbose use of the `actual` property. However, I can appreciate the desire for the explicit form in the context of Ainneve as an example for beginner programmers. Do you want me to take it all out?